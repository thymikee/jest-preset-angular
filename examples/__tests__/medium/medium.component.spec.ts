import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ChildComponent } from './child.component';
import { MediumComponent } from './medium.component';

describe('MediumComponent', () => {
  let component: MediumComponent;
  let fixture: ComponentFixture<MediumComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [MediumComponent, ChildComponent],
    })
      .overrideComponent(ChildComponent, { set: { template: 'stubbedBody' } })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MediumComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('snapshot test', () => {
    expect(fixture).toMatchSnapshot();
  });
});
