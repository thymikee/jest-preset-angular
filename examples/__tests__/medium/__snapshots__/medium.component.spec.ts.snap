// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediumComponent snapshot test 1`] = `
<app-medium
  __ngContext__={[Function LRootView]}
  anotherVar="false"
  someVar={[Function Boolean]}
>
  <div
    class="some-wrapper-class"
  >
     diary works! 
    <div
      class="inner-class"
    >
      
      <div>
        another case
      </div>
      <span>
        some html
      </span>
    </div>
    
    <div>
      one more case case
    </div>
    <div>
      <app-child-component
        someinput="someVar"
      >
        stubbedBody
      </app-child-component>
    </div>
  </div>
</app-medium>
`;
